src/DataValues/TimeValue.php

```diff
@@ -140,9 +140,10 @@ public function __construct( $time, $timezone, $before, $after, $precision, $cal
 		}
 
 		$year = $matches[2];
+		$year = ltrim( $year, '0' );
 		$year = str_pad( $year, 4, '0', STR_PAD_LEFT );
 
-		$this->time = $matches[1] . $year . $matches[3];
+		$this->timestamp = $matches[1] . $year . $matches[3];
```

Comment:
I disagree. Moving validation somewhere else means there will always be a code path that allows constructing invalid time values. Having this code here is the only way to be sure.